piwebapi_samples/Data Analysis/PIUploadUtility/PIUploadUtility/PIUploadUtility/Program.cs

```diff
@@ -0,0 +1,243 @@
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.InnerException.Message);
+                }
+            }
+        }
+
+        static bool IsTagExist(string dataserver)
+        {
+            string tagname = "VAVCO 2-09.Predicted Cooling Time";
+           
+            string path = "\\\\" + dataserver + "\\" + tagname;
+            string getPointQuery = "points?path=" + path;
+
+            try
+            {
+                JObject result = client.GetRequest(getPointQuery);
+            }
+            catch (Exception e)
+            {
+                if (e.InnerException.Message.Contains("404"))
+                {
+                    return false;
+                }
+                else
+                {
+                    Console.WriteLine(e.InnerException.Message);
+                }
+            }
+            
+            return true;
+        }
+        static void UpdateValues(string dataserver, string tagDefinitionLocation, string PIDataLocation)
+        {
+            var tags = File.ReadLines(tagDefinitionLocation);
+
+            foreach (string tag in tags)
+            {
+                string[] split = tag.Split(',');
+                string tagname = split[0];
+                List<string[]> entries = new List<string[]>();
+
+                var values = File.ReadLines(PIDataLocation);
+                foreach (string value in values)
+                {
+                    if (value.Contains(tagname))
+                    {
+                        entries.Add(value.Split(','));
+                    }
+                }
+
+                string path = "\\\\" + dataserver + "\\" + tagname;
+                string webid = GetWebIDByPath(path, "points");
+                string updateValueQuery = "streamsets/recorded";
+
+                List<Object> items = new List<Object>();
+                foreach (string[] line in entries)
+                {                    
+                    Object item = new
+                    {
+                        Timestamp = line[3],
+                        Value = line[1]
+                    };
+                    items.Add(item);
+                }
+
+                Object payload = new
+                {
+                    Items = items.ToArray(),
+                    WebId = webid
+                };
+
+                string request_body = "[" + JsonConvert.SerializeObject(payload) + "]";
+
+                try
+                {
+                    client.PostRequest(updateValueQuery, request_body);
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine(e.InnerException.Message);
+                }
+            }
+        }
+       
+        static void Main(string[] args)
+        {
+            //Use the default values provided above (which work when running from visual studio) or use the values provided by command line arguments
```

Comment:
At the top of this file. Saying "provided above" might not be clear because it's very far above.